code.rb

```diff
@@ -1,46 +1,51 @@
-@morse_code = {
-  'a' => '.-',
-  'b' => '-...',
-  'c' => '-.-.',
-  'd' => '-..',
-  'e' => '.',
-  'f' => '..-.',
-  'g' => '--.',
-  'h' => '....',
-  'i' => '..',
-  'j' => '.---',
-  'k' => '-.-',
-  'l' => '.-..',
-  'm' => '--',
-  'n' => '-.',
-  'o' => '---',
-  'p' => '.--.',
-  'q' => '--.-',
-  'r' => '.-.',
-  's' => '...',
-  't' => '-',
-  'u' => '..-',
-  'v' => '...-',
-  'w' => '.--',
-  'x' => '-..-',
-  'y' => '-.--',
-  'z' => '--..',
-  ' ' => ' ',
-  '1' => '.----',
-  '2' => '..---',
-  '3' => '...--',
-  '4' => '....-',
-  '5' => '.....',
-  '6' => '-....',
-  '7' => '--...',
-  '8' => '---..',
-  '9' => '----.',
-  '0' => '-----'
+@code = {
+  '.-' => 'a',
+  '-...' => 'b',
+  '-.-.' => 'c',
+  '-..' => 'd',
+  '.' => 'e',
+  '..-.' => 'f',
+  '--.' => 'g',
+  '....' => 'h',
+  '..' => 'i',
+  '.---' => 'j',
+  '-.-' => 'k',
+  '.-..' => 'l',
+  '--' => 'm',
+  '-.' => 'n',
+  '---' => 'o',
+  '.--.' => 'p',
+  '--.-' => 'q',
+  '.-.' => 'r',
+  '...' => 's',
+  '-' => 't',
+  '..-' => 'u',
+  '...-' => 'v',
+  '.--' => 'w',
+  '-..-' => 'x',
+  '-.--' => 'y',
+  '--..' => 'z'
 }
 
-def morse_decoder(string)
-    string.split(' ').map { |x| @morse_code.key(x) }.join
+def decode(string)
+  coded_words = string.split('   ')
+  coded_letters = []
+  letters = []
```

Comment:
- [ ] As mentioned by the previous reviewer and for the project requirements, kindly make sure you create 3 separate methods where:
  - The 1st decodes a Morse code character, takes a string parameter, and returns the corresponding character in uppercase (e.g.` decode_char(".-")` returns "A").
  - The 2nd decodes an entire word in Morse code, takes a string parameter, and returns the string representation. Every character in a word will be separated by a single space (e.g. `decode_word("-- -.--")` returns "MY").
  - The last one decodes the entire message in Morse code, takes a string parameter, and returns the string representation. Every word will be separated by 3 spaces (e.g. `decode("-- -.--   -. .- -- .")` returns "MY NAME").

I can suggest you structure your code like in the below example:
```ruby
morse_letters = {
    '.-' => 'a',
    '-...' => 'b',
    '-.-.' => 'c',
    '-..' => 'd',
    '.' => 'e',
    ...
 }

  def decode_char 
    # Code for the decode_char method here
  end

  def decode_word 
    # Code for the decode_word method here
  end

  def decode
    # Code for the decode method here
  end
```